### PR TITLE
chore: remove _even more_ code with TypeScript's `noUnusedParams`

### PR DESCRIPTION
**Pull request by danielgent**
Following on from https://github.com/sesamyab/auth/pull/419 but with the typescript config `noUnusedParams`

### CODE REVIEW

File: src/adapters/email/index.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import { sendCode, sendLink } from "../../controllers/email";
-import { Env } from "../../types";
 import { DataAdapters } from "../interfaces";
-export default function createEmailAdapter(env: Env): Partial<DataAdapters> {
+export default function createEmailAdapter(): Partial<DataAdapters> {
```
Comment (danielgent) [inline src/adapters/email/index.ts:3]: we pass the `env` in on each call

File: src/helpers/generate-auth-response.ts [hunk 2]
```diff
@@ -31,7 +31,6 @@ interface GenerateAuthResponseParamsForIdToken
 }
 
 async function generateCode({
-  env,
```
Comment (danielgent) [inline src/helpers/generate-auth-response.ts:34]: this is used in the other calls but not here

File: src/routes/management-api/keys.ts [hunk 3]
```diff
@@ -25,7 +24,6 @@ export class KeysController extends Controller {
   @Get("")
   public async list(
     @Request() request: RequestWithContext,
-    @Header("tenant-id") tenantId: string,
```
Comment (danielgent) [inline src/routes/management-api/keys.ts:28]: we don't use this header

File: test/fixtures/oauth2Client.ts [hunk 4]
```diff
@@ -22,10 +22,12 @@ class OAuth2ClientFixture implements IOAuth2Client {
     this.redirectUri = redirectUri;
   }
 
+  //@ts-ignore
```
Comment (danielgent) [inline test/fixtures/oauth2Client.ts:25]: Same as the previous PR. We need the same function signatures to match the mock

File: tsconfig.json [hunk 5]
```diff
@@ -26,7 +26,8 @@
     "paths": {
       "*": ["node_modules/*"]
     },
-    "noUnusedLocals": false
+    "noUnusedLocals": false,
+    "noUnusedParameters": false
```
Comment (danielgent) [inline tsconfig.json:30]: I did this PR by setting this to `true`. There's still a lot though!

File: src/routes/management-api/keys.ts (hunk 3 above)
Comment (markusahlstrand) [reply src/routes/management-api/keys.ts:28]: This is actually changing the swagger definition.. Think I would keep this one.

File: src/routes/management-api/keys.ts [hunk 6]
```diff
@@ -69,10 +66,7 @@ export class KeysController extends Controller {
 
   @Post("rotate")
   @SuccessResponse(201, "Created")
-  public async rotate(
-    @Request() request: RequestWithContext,
-    @Header("tenant-id") tenantId: string,
-  ): Promise<string> {
+  public async rotate(@Request() request: RequestWithContext): Promise<string> {
```
Comment (markusahlstrand) [inline src/routes/management-api/keys.ts:69]: Same here.. I would keep this one if we decide to do per tenant keys.

File: src/routes/management-api/keys.ts [hunk 7]
```diff
@@ -92,7 +86,6 @@ export class KeysController extends Controller {
   @SuccessResponse(201, "Created")
   public async revoke(
     @Request() request: RequestWithContext,
-    @Header() tenant_id: string,
```
Comment (markusahlstrand) [inline src/routes/management-api/keys.ts:95]: Same here.. I would keep this one if we decide to do per tenant keys.

File: src/server.ts [hunk 8]
```diff
@@ -25,18 +25,18 @@ const server = {
...
+          ...createEmailAdapter(),
           ...createAdapters(db),
           ...createR2Adapter(env),
         },
       },
       ctx,
     );
   },
-  async scheduled(event: ScheduledEvent, env: Env, ctx: ExecutionContext) {
+  async scheduled() {
```
[the remark this answers is not in the document]
Comment (markusahlstrand) [reply src/server.ts:36]: These are for cron events in cloudflare. I'm thinking that we'll add a nightly event that will prune the logs a maybe rotate the keys.

File: src/server.ts [hunk 9]
```diff
@@ -25,18 +25,18 @@ const server = {
...
     // Rotate keys and trim tables
   },
-  async queue(batch: MessageBatch, env: Env, ctx: ExecutionContext) {
+  async queue() {
```
Comment (markusahlstrand) [inline src/server.ts:39]: We can remove this entire function as we're not using the queue

**Review by markusahlstrand — COMMENTED** [review]
This is really cool. I think we need to be a bit careful when patching anything that is generating TSOA routes though

File: src/routes/management-api/keys.ts (hunk 3 above)
Comment (danielgent) [reply src/routes/management-api/keys.ts]: Cool ok. I've reverted this. We are not actually using the header though (thus why I could remove it :wink: )

As long as you know this!

**Comment by danielgent** [comment]
Cool ok @markusahlstrand . I've reverted the changes to the keys mgmt API route

Nothing in here is urgent so if you've lost track we could go through it in a videocall :video_camera:

**Review by markusahlstrand — APPROVED** [review]
Cool! Spring cleaning!

**Pull request merged**